### PR TITLE
add terra mat paginator intl

### PR DESCRIPTION
**Pull request by ChristianMoll**
@plentymarkets/team-terra

#### Must be done by Terra

Testing
- [x] Person 1 (mandatory)

    Browser-Support (relevant for changes in scss / appearance of components)
    - [x] Chrome
    - [x] Firefox
    - [x] Safari
    
    Responsiveness
    - [x] Desktop
    - [x] Tablet
    - [x] Mobile
    
- [x] Person 2 (mandatory)

    Browser-Support
    - [x] Chrome
    - [x] Firefox
    - [x] Safari
    
    Responsiveness
    - [x] Desktop
    - [x] Tablet
    - [x] Mobile

Information transfer
- [ ] Inform Ceres about changes in `TerraFormComponent` (optional: relevant for changes in this component)

Terra Basic Plugin
- [ ] Adapt changes (optional: relevant for version updates, global design changes, l10n)

----

#### Must be done by every developer
Please inform a member of Terra (@plentymarkets/team-terra) to get the upper part of the checklist done (with urgency or deadline). 

Documentation
- [x] Changelog (optional: relevant for every change which influences the API)
- [x] Example (updated or created)
- [x] JSDoc (optional: relevant for every change which influences the API)
- [x] [Google spreadsheet](https://docs.google.com/spreadsheets/d/1OINnux8TEoitV-qdAxqUQaf9oGI_fqFVwfRWenRFfhI/edit#gid=0) (relevant when deprecating public APIs or features)

Testing
- [ ] Unit Test (updated or created)

### CODE REVIEW

File: src/lib/components/table/terra-mat-paginator-intl.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        });
+    }
+
+    public getRangeLabel = (page:number, pageSize:number, length:number):string =>
+    {
+        let ofLabel:string = this.translation.translate('terraMatPaginatorIntl.ofLabel');
+
+        if(length === 0 || pageSize === 0)
+        {
+            return `0 / ${length}`;
```
Comment (MFrank84) [inline src/lib/components/table/terra-mat-paginator-intl.ts:26]:
```suggestion
            return `0 ${ofLabel} ${length}`;
```

File: src/assets/lang/locale-en.json [hunk 2]
```diff
@@ -130,5 +130,13 @@
 		"unauthorized": "Unauthorized",
 		"missingPermission": "Missing permissions for:",
 		"forbidden": "Forbidden"
+	},
+	"terraMatPaginatorIntl": {
+		"itemsPerPage": "Items per page",
```
Comment (johannabecker) [inline src/assets/lang/locale-en.json:135]:
```suggestion
		"itemsPerPage": "Results per page",
```

File: src/lib/components/table/index.ts [hunk 3]
```diff
@@ -4,3 +4,4 @@ export { TablePagingDataSource } from './table-paging-data-source';
 export { TableSortingDataSource } from './table-sorting-data-source';
 export { TablePagingSortingDataSource } from './table-paging-sorting-data-source';
 export { RequestParameterInterface } from './request-parameter.interface';
+export { TerraMatPaginatorIntl } from './terra-mat-paginator-intl';
```
Comment (Neill83) [inline src/lib/components/table/index.ts:7]: dir fehlt noch der export in der public-api.ts

File: src/assets/lang/locale-en.json [hunk 4]
```diff
@@ -130,5 +130,13 @@
 		"unauthorized": "Unauthorized",
 		"missingPermission": "Missing permissions for:",
 		"forbidden": "Forbidden"
+	},
+	"terraMatPaginatorIntl": {
+		"itemsPerPage": "Results per page",
+		"nextPage": "",
```
Comment (Neill83) [inline src/assets/lang/locale-en.json:136]: die "leeren" strings alle löschen

File: src/lib/components/table/terra-mat-paginator-intl.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        length = Math.max(length, 0);
+        const startIndex:number = page * pageSize;
+        const endIndex:number = startIndex < length ? Math.min(startIndex + pageSize, length) : startIndex + pageSize;
+        return `${startIndex + 1} – ${endIndex} ${ofLabel} ${length}`;
+    }
+
+    private updateLabels():void
+    {
+        this.itemsPerPageLabel = this.translation.translate('terraMatPaginatorIntl.itemsPerPage');
+        this.nextPageLabel = this.translation.translate('terraMatPaginatorIntl.nextPage');
```
Comment (Neill83) [inline src/lib/components/table/terra-mat-paginator-intl.ts:35]: und hier einfach `this.nextPageLabel = '';` überall

File: src/lib/components/table/terra-mat-paginator-intl.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        return `${startIndex + 1} – ${endIndex} ${ofLabel} ${length}`;
+    }
+
+    private updateLabels():void
+    {
+        this.itemsPerPageLabel = this.translation.translate('terraMatPaginatorIntl.itemsPerPage');
+        this.nextPageLabel = '';
+        this.previousPageLabel = '';
+        this.firstPageLabel = '';
+        this.lastPageLabel = '';
```
Comment (MFrank84) [inline src/lib/components/table/terra-mat-paginator-intl.ts:38]:
```suggestion
```

Der leere String sollte nicht immer wieder zugewiesen werden.

File: src/lib/components/table/terra-mat-paginator-intl.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+import { MatPaginatorIntl } from '@angular/material';
+import { TranslationService } from 'angular-l10n';
+
+@Injectable()
+export class TerraMatPaginatorIntl extends MatPaginatorIntl
+{
+    constructor(private translation:TranslationService)
```
Comment (MFrank84) [inline src/lib/components/table/terra-mat-paginator-intl.ts:8]:
```suggestion
    public nextPageLabel:string = '';
    public previousPageLabel:string = '';
    public firstPageLabel:string = '';
    public lastPageLabel:string = '';
        
    constructor(private translation:TranslationService)
```

Damit der leere String nicht immer wieder zugewiesen werden muss.

**Pull request merged**